crypto/err/err.c

```diff
@@ -584,7 +581,12 @@ char *ERR_error_string(unsigned long e, char *ret)
 
 LHASH_OF(ERR_STRING_DATA) *ERR_get_string_table(void)
```

Comment:
This function's threading is rather unclear, read or write lock. It's not actually safe for the caller to do anything with the result, especially since `err_string_lock` isn't exposed.

Should this race with, say, a `err_load_strings` call, it wouldn't be safe to iterate over the table or anything. It seems this function is only called, within OpenSSL, by `apps/errstr.c` which probably just assumes there's only one thread.